Automattic/wp-calypso
# Hide sidebar when Calypso is viewed from WC apps

#### Changes proposed in this Pull Request

* Using the `isWcMobileApp` utility method introduced in https://github.com/Automattic/wp-calypso/pull/54181, hide the sidebar if Calypso is accessed from the WooCommerce app (WPiOS or WCAndroid).

| Before | After |
| - | - |
| <img width="400" alt="Before: the sidebar is shown" src="https://user-images.githubusercontent.com/1898325/124031022-ccef8a00-d9c4-11eb-8a1a-77b5e4824a58.png"> | <img width="400" alt="After: the sidebar is hidden" src="https://user-images.githubusercontent.com/1898325/124031036-d1b43e00-d9c4-11eb-8d15-116a7de513d4.png"> |

**Note: This PR depends on https://github.com/Automattic/wp-calypso/pull/54181 which should be merged first.**

#### Testing instructions



* Checkout this branch and run the app locally: `yarn && yarn start`
* Open any Calypso page at http://calypso.localhost:3000 in the browser
* Verify the sidebar is visible as usual
* Change the user agent to that of WCiOS: `Mozilla/5.0 (iPhone; CPU iPhone OS 12_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/16B91 wc-ios/12.1`
* Reload the page and verify that the sidebar is hidden
* Change the user agent to that of WCAndroid: `Mozilla/5.0 (iPhone; CPU iPhone OS 12_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/16B91 wc-android/12.1`
* Reload the page and verify that the sidebar is still hidden

Tip: To change the user agent in Chrome, see below screenshots (taken from https://github.com/Automattic/wp-calypso/pull/31830):

<img width="404" alt="55129228-ab562700-5159-11e9-9a75-d05cfe97b65e" src="https://user-images.githubusercontent.com/1898325/124031455-556e2a80-d9c5-11eb-9a74-02a5315f4bf9.png">
<img width="900" alt="55129288-df314c80-5159-11e9-8278-f14471c9eb2f" src="https://user-images.githubusercontent.com/1898325/124031451-54d59400-d9c5-11eb-9924-d252d408879f.png">



Fixes https://github.com/Automattic/wp-calypso/issues/53982
